rpc/types/src/misc/misc.rs

```diff
@@ -0,0 +1,539 @@
+        distribution: Vec<u64>,
+        start_height: u64,
+        base: u64,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        1016..=1027
+    )]
+    /// Used in [`crate::json::GetMinerDataResponse`].
+    ///
+    /// Note that this is different than [`crate::misc::TxBacklogEntry`].
+    GetMinerDataTxBacklogEntry {
+        id: String,
+        weight: u64,
+        fee: u64,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        1070..=1079
+    )]
+    /// Used in [`crate::json::AddAuxPowRequest`].
+    AuxPow {
+        id: String,
+        hash: String,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        192..=199
+    )]
+    /// Used in [`crate::bin::GetBlocksResponse`].
+    TxOutputIndices {
+        indices: Vec<u64>,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        201..=208
+    )]
+    /// Used in [`crate::bin::GetBlocksResponse`].
+    BlockOutputIndices {
+        indices: Vec<TxOutputIndices>,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        210..=221
+    )]
+    /// Used in [`crate::bin::GetBlocksResponse`].
+    PoolTxInfo {
+        tx_hash: [u8; 32],
+        tx_blob: String,
+        double_spend_seen: bool,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "cryptonote_protocol/cryptonote_protocol_defs.h",
+        121..=131
+    )]
+    /// Used in [`crate::bin::GetBlocksResponse`].
+    TxBlobEntry {
+        blob: String,
+        prunable_hash: [u8; 32],
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        512..=521
+    )]
+    #[derive(Copy)]
+    ///
+    /// Used in:
+    /// - [`crate::bin::GetOutsRequest`]
+    /// - [`crate::other::GetOutsRequest`]
+    GetOutputsOut {
+        amount: u64,
+        index: u64,
+    }
+}
+
+define_struct_and_impl_epee! {
+    #[doc = monero_definition_link!(
+        cc73fe71162d564ffda8e549b79a350bca53c454,
+        "rpc/core_rpc_server_commands_defs.h",
+        538..=553
+    )]
+    #[derive(Copy)]
+    /// Used in [`crate::bin::GetOutsRequest`].
+    OutKeyBin {
+        key: u8, // TODO: crypto::public_key,
+        mask: u8, // TODO: rct::key,
```

Comment:
These are `[u8; 32]`

Reply:
These were going to be replace with some dedicated types and/or aliases as there's a few more of these (e.g. [`typedef std::string blobdata`](https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/cryptonote_basic/blobdatatype.h#L39)), but maybe the type directly is okay?